pennylane_lightning/lightning_kokkos/_state_vector.py

```diff
@@ -193,15 +192,24 @@ def _apply_lightning_controlled(self, operation):
         """
         state = self.state_vector
 
+        basename = operation.base.name
+        method = getattr(state, f"{basename}", None)
         control_wires = list(operation.control_wires)
         control_values = operation.control_values
-        name = operation.name
-        # Apply GlobalPhase
-        inv = False
-        param = operation.parameters[0]
-        wires = self.wires.indices(operation.wires)
-        matrix = global_phase_diagonal(param, self.wires, control_wires, control_values)
-        state.apply(name, wires, inv, [[param]], matrix)
+        target_wires = list(operation.target_wires)
+        inv = False  # TODO: update to use adjoint in C++ instead of in Python
```

Comment:
Where's the logic for applying controlled gates with `inv = True`?

Reply:
I have created a [ticket](https://app.shortcut.com/xanaduai/story/79430/apply-lightning-operations-recursively-for-controlled-adjoint-operations) to update all backends (LQ/LK/LGPU) together by applying operations recursively. This will allow us to deal with nested ops like `qml.ctrl(qml.adjoint(...` or `qml.adjoint(qml.ctrl(...` or `qml.ctrl(qml.ctrl(...` more effectively.

For now I will keep this implementation and update them together in the future.